README.md

```diff
@@ -361,11 +308,206 @@ All the current BigQuery datatypes are being handled when transforming data from
+* If a data record cannot be serialized by BigQuery sink, then the record is dropped with a warning getting logged. Moving on,
+  we plan to introduce a Flink metric for tracking such data. Additionally, a dead letter queue will be introduced in the future
+  to store this data.
+
+**Important:** Please refer to [data ingestion pricing](https://cloud.google.com/bigquery/pricing#data_ingestion_pricing) to
+understand the BigQuery Storage Write API pricing.
+
+### Table API Support
+ * Table API is a high-level declarative API that allows users to describe what they want to do 
+rather than how to do it. 
+ * This results in simpler customer code and higher level pipelines that are more easily optimized in a managed service.
+* The Table API is a superset of the SQL language and is specially designed for working with Apache Flink.
+* It also allows language-embedded style support for queries in Java, Scala or Python besides the always available String values as queries in SQL.
+  
+#### Source: Bounded and Unbounded
+```java
+// Note: Users must create and register a catalog table before reading and writing to them.
+
+// final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+// env.enableCheckpointing(CHECKPOINT_INTERVAL);
+// final StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+// Create the Config.
+BigQueryTableConfig readTableConfig =  new BigQueryReadTableConfig.Builder()
+        .table(...)
+        .project(...)
+        .dataset(...)
+        .partitionDiscoveryInterval(...)
+        .boundedness(...) // Boundedness.CONTINUOUS_UNBOUNDED or Boundedness.BOUNDED
+        .limit(...)
+        .columnProjection(...)
+        .snapshotTimestamp(...)
+        .oldestPartitionId(...)
+        .maxStreamCount(...)
+        .rowRestriction(...)
+        .build();
+
+// Create the catalog table.
+tEnv.createTable(
+        "bigQuerySourceTable",
+         BigQueryTableSchemaProvider.getTableDescriptor(readTableConfig));
+Table sourceTable = tEnv.from("bigQuerySourceTable");
+
+// Fetch entries in this sourceTable
+sourceTable = sourceTable.select($("*"));
+```
+#### Sink: At-least Once
+```java
+// Note: Users must create and register a catalog table before reading and writing to them.
+// Schema of the source and sink catalog table must be the same
+
+// final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+// env.enableCheckpointing(CHECKPOINT_INTERVAL);
+// final StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+// Create the Config.
+BigQueryTableConfig sinkTableConfig = BigQuerySinkTableConfig.newBuilder()
+        .table(...)
+        .project(...)
+        .dataset(...)
+        .testMode(...)
+        .credentialAccessToken(...)
+        .credentialFile(...)
+        .credentialKey(...)
+        .sinkParallelism(...) // Should be atmost 128
+        .deliveryGuarantee(...)
+        .build();
```

Comment:
Please mention "required" or "optional" against these parameters. Also, add a point (with link) to refer to table of sink configs for more details. Same applicable to source as well